codec-http/src/main/java/io/netty/handler/codec/http/HttpContentEncoder.java

```diff
@@ -73,9 +76,15 @@ public boolean acceptOutboundMessage(Object msg) throws Exception {
     @Override
     protected void decode(ChannelHandlerContext ctx, HttpRequest msg, List<Object> out)
             throws Exception {
-        CharSequence acceptedEncoding = msg.headers().get(HttpHeaderNames.ACCEPT_ENCODING);
-        if (acceptedEncoding == null) {
+        CharSequence acceptedEncoding;
+        List<String> acceptEncodingHeaders = msg.headers().getAll(ACCEPT_ENCODING);
```

Comment:
It's a shame that `HttpHeaders` doesn't have `get`/`getAll` methods which return `CharSequence`s rather than `String`s. These would be quite easy to add I think and would help cut down on copying and garbage.

(this is a passing comment and not directly related to the PR change!)